### ISSUE TITLE
Deb release?

### ISSUE THREAD

**Issue by simlu** [taffybar/taffybar#300]
I'm not very experiences with this, so what needs to be done to get a deb file of the latest version up [here](https://pkgs.org/download/taffybar)?

Unfortunately the latest version available 4.6 doesn't seem to work nicely with ubuntu 18.04 lts

**Comment by IvanMalison** [comment]
While we would welcome a contribution on this front. My position on this will always be that taffybar should be installed with stack or cabal, or from source. Generally speaking, installations from distributions have been more brittle and they have been slow to update. I don't really see what value they add either, except I suppose it avoids having to use a dedicated package manager. The thing is that most users of taffybar are going to want to be able to add additional packages and otherwise control things.

tl;dr If someone wants to make a deb package thats fine by me, but it will never be officially maintained or recommended as a way to install taffybar.

**Comment by simlu** [comment]
@IvanMalison I'm using ubuntu minimal cd version, and my problem is that it is really hard and inconvenient to use either one of the installation processes you described.

I need to install something like 2gb of dependencies to be able to install taffybar from cabal, while using deb I can just install it and it works. Just to put this in perspective this increases my disk usage for a blank install by roughly 500 percent.

It's

```shell
sudo apt-get install xmonad
sudo apt-get install libiw30
sudo dpkg -i taffybar_x.x.x-x_amd64.deb
```

vs

```
sudo apt-get install cabal-install
sudo apt-get install -f
sudo apt-get install cabal-install
sudo apt-get install libgtk-3-dev
sudo apt-get install xorg openbox
sudo apt-get install libgirepository1.0-dev libwebkit2gtk-4.0-dev libgtksourceview-3.0-dev
sudo apt-get install libasound2 libasound2-dev
sudo apt-get install c2hs
cabal update
cabal install cabal-install
cabal install alex happy
cabal install xmonad
cabal install taffybar
```

I wasn't able to use the build script at all since I couldn't figure out which dependencies were needed and it took me quite a while to understand what was required for cabal. 

Taffybar is designed to be a minimalists status bar. Having to install gigabytes of dependencies isn't great when you're trying to build a minimal installation.

So that's my reasoning why I would really like a ***.deb**

Anyone here knows what it takes to build one? I guess an alternative is switching back to xmobar, however I really like taffybar and have heavily customized it to my needs.

Thank you for any input!

**Comment by IvanMalison** [comment]
> I need to install something like 2gb of dependencies

All of the non-haskell dependencies that are needed for the cabal install will also be needed for the .deb, so I don't think your comparison is fair at all. Whether you use a .deb or not, you are going to need ghc installed, since taffybar depends on ghc to recompile your config. taffybar also needs most of its dependencies at configuration compilation time anyway. I'm not really sure how exactly its working with the current .deb situation, but I seriously doubt that disk usage is significantly different.

 Let me go through all of these:

```
sudo apt-get install libgtk-3-dev # Needed either way
sudo apt-get install xorg openbox # Not sure where you are getting this from
sudo apt-get install libgirepository1.0-dev libwebkit2gtk-4.0-dev libgtksourceview-3.0-dev # Needed either way
sudo apt-get install libasound2 libasound2-dev # Needed either way
sudo apt-get install c2hs
```

taffybar has added some dependencies lately.

I'm not an expert in debian packaging, but I expect that 

sudo dpkg -i taffybar_x.x.x-x_amd64.deb

takes care of installing dependencies for you, does it not?

In any case, if you are merely worried about the number of directives you need to issue to get things going, I would recommend using stack.

You can take a peek at how to do that in the quick-start script, or better yet, just let the quick-start script do everything for you.

**Comment by LeifW** [comment]
You'd be better off asking in a Debian-related forum. E.g. a mailing list for Debian packages, or a Haskell-specific subset of that, such as the "Debian Haskell Group".
If you go to the page for the package: https://packages.debian.org/stretch/taffybar
and click on the "Developer Information" link, you'll see that they already have a ticket open to update the package to version 1.0.2.
There's also a source link in there with the source code for the project, with a "debian" folder added, containing package description files.

**Comment by simlu** [comment]
@IvanMalison Thank you for your detailed response! I'm comparing virtual machine disk size between the two installations. So implicit or explicit install of dependencies shouldn't make a difference.

`taffybar has added some dependencies lately.` - so this might be the difference in the size that I'm seeing then?

`dpkg -i ...` installs some dependencies, but not nearly as much as when using the cabal install - however I haven't tried installing `0.4.6` using cabal - so the difference might be much smaller

Unfortunately using the quick-start script didn't work. I can take another step at it later and see if I can figure out which dependencies are missing.

@LeifW Thank you very much for the reply! Thank link looks very promising. I'll try to do some digging

**Comment by IvanMalison** [comment]
@simlu The debian package seems to be on version 3.0.0 That should be good enough, right?

**Comment by simlu** [comment]
Maybe. I'll try again soon ;)
